### PR TITLE
Renamed unsafe lifecycle components

### PR DESCRIPTION
**Pull request by helloitsm3**

### CODE REVIEW

**Comment by zpao** [comment]
Will take a look at this soon. Need to figure out what the compatibility story is - right now we technically support React 15, though that's surely unnecessary. We also don't properly support React 17 either since the unprefixed method in use now doesn't run. May leave as is, then take this and call it 2.0 and only support 16.3+.

File: src/index.js [hunk 1]
```diff
@@ -230,11 +230,12 @@ class QRCodeCanvas extends React.PureComponent<QRProps, {imgLoaded: boolean}> {
...
   }
 
-  componentWillReceiveProps(nextProps) {
+  static getDerivedStateFromProps(nextProps, prevState) {
     const currentSrc = this.props.imageSettings?.src;
     const nextSrc = nextProps.imageSettings?.src;
+
     if (currentSrc !== nextSrc) {
-      this.setState({imgLoaded: false});
+      this.setState({imgLoaded: false})
```
Comment (jcvidiri) [inline src/index.js:238]: I think it should be renamed to `UNSAFE_componentWillReceiveProps` as its recommended in the [official update post](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html). Also, please don't remove the semicolon after the `setState()`

**Review by jcvidiri — CHANGES_REQUESTED** [review]

**Comment by helloitsm3** [comment]
I've updated to using componentDidUpdate. Do take a look

File: src/index.js (hunk 1 above)
Comment (helloitsm3) [reply src/index.js]: I've updated it to use componentDidUpdate instead. Do take a look

Comment (artola) [reply src/index.js]: @zpao Surely you know better than me that the suggested change to `getDerivedStateFromProps ` is not doable as written.
But we can store the `prevProps` in state and derive the state accordingly when new props arrive:

```js
   static getDerivedStateFromProps(props, state) {
    const currentSrc = state.imageSrc;
    const nextSrc = props.imageSettings?.src;

    if (currentSrc !== nextSrc) {
      return {imgSrc: nextSrc, imgLoaded: false};
    }
  }
```

Could you pls publish some fix for this issue? using the mentioned above approach or just renaming the method `UNSAFE_componentWillReceiveProps` at least for the >= v16.3.0.

**Comment by zpao** [comment]
Shipped v2 with use of `unsafe_componentWillReceiveProps` instead.

**Pull request closed without merging**